eclipse/src/saros/project/SharedResourcesManager.java

```diff
@@ -83,30 +84,7 @@
    */
   @Inject private FileReplacementInProgressObservable fileReplacementInProgressObservable;
 
-  private final ProjectDeltaVisitor projectDeltaVisitor;
-
-  /** map that holds the current open or closed state for every shared project */
-  private final Map<IProject, Boolean> projectStates = new HashMap<IProject, Boolean>();
-
-  private final ISessionListener sessionListener =
-      new ISessionListener() {
-
-        @Override
-        public void referencePointAdded(IReferencePoint referencePoint) {
```

Comment:
Any reason to not use this listeners to update a map that you just calculate over and over again in createProjectReferencePointMap ?

Reply:
I would like to avoid caching this mapping as updating it is not always that easy. I am not sure how Eclipse handles internal renamings/moves (i.e. whether the existing resource objects for the changed resources are updated or whether they are just invalidated).

Furthermore, a user could theoretically move a reference point to a different project. (I know this is currently not strictly supported by Saros/E, but still. Also, I don't see a reason why to not support this case. With the new reference point implementation, renaming/moving a reference point or one of its parents should not be an issue if resource objects are updated internally for such changes. This is already supported by Saros/I.)

Also, since this is only done once per resource delta, creating the mapping in O(#ref-points) is acceptable in my opinion.